.github/workflows/portable-rustls-ci-build.yml

```diff
@@ -0,0 +1,201 @@
+  pull_request:
+  merge_group:
+  schedule:
+    - cron: '0 18 * * *'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: build with unstable_portable_atomic_arc
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        rust:
+          - nightly
+          - nightly-2024-01-01
+          # RUST NIGHTLY version(s) for MSRV - 1.71:
+          - nightly-2023-06-01
+          # FAILS AT: cargo add once_cell
+          # - nightly-2023-05-01
+        os:
+          - ubuntu-latest
+          # FOR FUTURE CONSIDERATION:
+          # - macos-latest
+          # - windows-latest
+        target:
+          # 32-BIT TARGET(S) WITH NO ATOMIC PTR:
+          - riscv32imc-unknown-none-elf
+          # 64-bit target(s) with no std crate - for comparison:
+          - aarch64-unknown-none
+          - x86_64-unknown-none
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Install ${{ matrix.rust }} toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.rust }}
+          target: ${{ matrix.target }}
+
+      # TODO: CARGO FEATURE FOR THIS STEP
+      - name: additional dependencies as needed for target(s) with no atomic ptr
+        run: |
+          cargo add once_cell -F critical-section -p portable-rustls
+          cargo add portable-atomic -F critical-section -p portable-rustls
+
+      - name: cargo build (debug; no default features; no-std) # (with no built-in provider)
+        run: cargo build --target ${{ matrix.target }}
+        working-directory: rustls
+        env:
+          RUSTFLAGS: --cfg portable_atomic_unstable_coerce_unsized --cfg unstable_portable_atomic_arc
+
+      # NOTE: BUILD with hashbrown FAILS with no atomic ptr - MISSING SUPPORT in default hasher
+      # - name: cargo build (debug; no default features; no-std; hashbrown) # (with no built-in provider)
+      #   ...
+
+  test:
+    # GENERAL NOTE: NO CI TESTING for no-std
+    # (CI TESTING with no-std & with no atomic ptr FOR FUTURE CONSIDERATION)
+    name: test with unstable_portable_atomic_arc
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        rust:
+          - nightly
+          - nightly-2024-05-01
+          # FAILS cargo test: use of unstable library feature in proc-macro2
+          # - nightly-2024-04-01
+        os:
+          - ubuntu-latest
+          - macos-latest
+          # FOR FUTURE CONSIDERATION:
+          # - windows-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Install ${{ matrix.rust }} toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.rust }}
+          target: ${{ matrix.target }}
```

Comment:
should have removed `target: ${{ matrix.target }}` from here as there is no `target` in the matrix for this test job and there is no need to add any extra, special Rust build target for this test job.